libraries/apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/DefaultApolloStore.kt

```diff
@@ -36,7 +36,11 @@ internal class DefaultApolloStore(
 
   override val changedKeys = changedKeysEvents.asSharedFlow()
 
-  private val cache: OptimisticCache = OptimisticCache().chain(normalizedCacheFactory.createChain()) as OptimisticCache
+  // Keeping this as lazy to avoid accessing the disk at initialization which usually happens on the main thread
+  private val cache: OptimisticCache by lazy {
+    OptimisticCache().chain(normalizedCacheFactory.createChain()) as OptimisticCache
```

Comment:
Is lazy thread safe? I remember reading something or some rumour saying `lazy` was generating a lot of bytecode. Might be worth making it a lateninit and use regular locks?

Reply:
By default `by lazy` uses a [synchronized](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/jvm/src/kotlin/util/LazyJVM.kt#L69) block on the JVM and on native it [looks](https://github.com/JetBrains/kotlin/blob/master/kotlin-native/runtime/src/main/kotlin/kotlin/native/concurrent/Lazy.kt#L16) like it's using a Lock.

For the bytecode yes it can be "verbose" (👀  [this](https://youtu.be/Ta5wBJsC39s?t=1173) from 2019!)

Pushed a commit doing this manually LMKWYT!

Reply:
Looks probably ok to keep `lazy` here according to Chet guidance. It's quite an heavy object and allocation is done only once so both work for me!